src/main/java/org/opentripplanner/middleware/controllers/api/OtpRequestProcessor.java

```diff
@@ -122,9 +128,31 @@ private static void handlePlanTripResponse(Request request, OtpDispatcherRespons
         long tripStorageStartTime = DateTimeUtils.currentTimeMillis();
 
         Auth0Connection.checkUser(request);
-        Auth0UserProfile profile = Auth0Connection.getUserFromRequest(request);
+        RequestingUser requestingUser = Auth0Connection.getUserFromRequest(request);
+        // A requesting user (Otp or third party user) is required to proceed.
+        if (requestingUser == null) {
+            return;
+        }
+
+        // Determine if the Otp request is being made by an actual Otp user or by a third party on behalf of an Otp user.
+        OtpUser otpUser = null;
+        if (requestingUser.otpUser != null) {
```

Comment:
FYI there is also an edge case here that, if an `ApiUser` is also an `OtpUser`, they will be able to make OTP requests on behalf of any `OtpUser`.

Reply:
@binh-dam-ibigroup I'm not sure that is the case. `Auth0Connection.checkUser(request)` will place authenticated users in the request based on the provided token. At this point, only users assoicated with the requesting user's token would be available. Unless I'm missing something here?

Reply:
Yes. In the case I am highlighting, `requestingUser.OtpUser` is populated, so that condition clears regardless of the association of the `userId` parameter with any ApiUser, and the request will be forwarded to OTP.

Reply:
@binh-dam-ibigroup by this point the response from OTP is being processed (not being forwarded to OTP). Prior to my latest update the plan was being saved against the ApiUser's matching OtpUser and not the intended OtpUser referenced by the userId. Inline with this, I have updated the unit test to create an OtpUser matching the ApiUser.

Reply:
Hmm intuitively you would instead make these checks before sending the OTP requests. I'm ok with how it is now, but please place a TODO comment about considering moving the checks to prior to sending the OTP request.

Reply:
I have moved the checks to before sending the request to OTP.